File: 0011.盛最多水的容器/0011-盛最多水的容器.rs
impl Solution {
    pub fn max_area(xs: Vec<i32>) -> i32 {
        let mut i = 0;
        let mut j = xs.len()-1;
        let mut ret = 0;
        while i < j {
            ret = ret.max(xs[i].min(xs[j]) * ((j-i) as i32));
            if xs[i] < xs[j] {
                i += 1;
            } else {
                j -= 1;
            }
        }
        
        ret
    }
}

File: 0011.盛最多水的容器/0011-盛最多水的容器.py
class Solution:
    def maxArea(self, nums: List[int]) -> int:
        i, j = 0, len(nums) - 1
        ret = 0
        while i < j:
            ret = max(ret, min(nums[i], nums[j]) * (j - i))
            if nums[i] < nums[j]:
                i += 1
            else:
                j -= 1
        return ret
